backbone/src/main/java/org/researchstack/backbone/task/Task.java

```diff
@@ -40,6 +46,53 @@ public Task(String identifier)
     }
 
     /**
+     * Sets the current @{@link ViewTaskActivity} for this task.
+     * This function should be called only by the {@link ViewTaskActivity} instance running
+     * at its creation time.
+     * @param activity the @{@link ViewTaskActivity} instance
+     */
+    public void setActivity(ViewTaskActivity activity){
```

Comment:
Thanks for your input, that makes sense and timed walk is certainly something RS needs to support. Definitely agree a rethink is needed and is something we will be tackling.